src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -404,49 +405,65 @@ object PhaseBlock extends LazyLogging {
       progress.record(ctx.getContig, ctx.getStart)
     }
 
+    val blockSet = new util.TreeSet[PhaseBlock](new Comparator[PhaseBlock] {
```

Comment:
Since it has `pollFirst`, which retrieves the first element and removes it all in one shot!